horovod/run/common/util/network.py

```diff
@@ -161,8 +161,16 @@ def __init__(self, service_name, addresses, key, match_intf=False,
         self._addresses = self._probe(addresses)
         if not self._addresses:
             raise NoValidAddressesFound(
-                'Unable to connect to the %s on any of the addresses: %s'
-                % (service_name, addresses))
+                'Horovodrun was unable to connect to the {service_name} on any '
+                'of the addresses: {addresses}. One of situations that can '
```

Comment:
Suggestion - can you make a separate paragraph for "One of situations" and everything afterward?  Paragraph = `\n\n`.